src/components/Tokens/TokenDetails/ChartSection.tsx

```diff
@@ -87,6 +87,24 @@ export default function ChartSection({
 
   const logoSrc = useTokenLogoURI(token, nativeCurrency)
 
+  // Backend doesn't always return latest price point for every duration.
+  // Thus we need to manually determine latest price point available, and
+  // append it to the prices list for every duration.
+  let latestPricePoint: PricePoint = { value: 0, timestamp: 0 }
+  let latestPricePointTimePeriod: TimePeriod
+  Object.keys(prices).forEach((key) => {
```

Comment:
could you memoize this calculation? you're creating a new object every render

Reply:
big OOF thanks for catching done